src/group_theory/commensurable.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2021 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import group_theory.index
+import group_theory.quotient_group
+import group_theory.subgroup.pointwise
+import group_theory.group_action.conj_act
+/-!
+# Commensurability for subgroups
+
+This file defines commensurability for subgroups of a group `G`. It then goes on to prove that
+commensurability defines an equivalence relation and finally defines the commensurator of a subgroup
+of `G`.
+
+## Main definitions
+
+* `commensurable`: defines commensurability for two subgroups `H`, `K` of  `G`
+* `commensurator`: defines the commensurator of a subgroup `H` of `G`.
+-/
+
+variables {G : Type*} [group G]
+
+/--Two subgroups `H K` of `G` are commensurable if `H ⊓ K` has finite index in both `H` and `K` -/
+def commensurable (H K : subgroup G) : Prop :=
+H.relindex K ≠ 0 ∧ K.relindex H ≠ 0
+
+namespace commensurable
+
+open_locale pointwise
+
+@[refl] protected lemma refl (H : subgroup G) : commensurable H H := by
+ simp only [commensurable, nat.one_ne_zero, ne.def, not_false_iff, and_self, subgroup.relindex_self]
```

Comment:
```suggestion
@[refl] protected lemma refl (H : subgroup G) : commensurable H H :=
simp [commensurable]
```

Reply:
What are the rules for `simp`? When should I squeeze it when shouldn't I?

Reply:
Aha ok cool. Thanks!

Reply:
The reason is simple: if for example you have `simp, rw [...]` and in the future someone add a `simp` lemma changing the behavior of the first `simp`, then the `rw` could not work anymore and your proof is broken. On the other way, if `simp` closes a goal, it's unlikely (impossible at least in theory?) that adding new `simp` lemmas can break things.